presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushProjectionIntoTableScan.java

```diff
@@ -164,4 +170,49 @@ public Result apply(ProjectNode project, Captures captures, Context context)
                                 newScanAssignments),
                         newProjectionAssignments.build()));
     }
+
+    private Optional<PlanNode> pushAssignmentsIntoTableScan(ProjectNode project, TableScanNode tableScan)
+    {
+        if (project.isIdentity()) {
+            // Can be removed by RemoveRedundantIdentityProjections
+            return Optional.empty();
+        }
+
+        if (project.getOutputSymbols().size() != tableScan.getOutputSymbols().size()) {
+            // Can be simplified with PruneTableScanColumns
+            return Optional.empty();
+        }
+
+        Map<Symbol, ColumnHandle> assignments = tableScan.getAssignments();
+
+        Assignments.Builder newProjectionAssignments = Assignments.builder();
+        Set<Symbol> newScanOutputs = new LinkedHashSet<>();
+        ImmutableMap.Builder<Symbol, ColumnHandle> newScanAssignments = ImmutableMap.builder();
+
+        for (Map.Entry<Symbol, Expression> projectAssignment : project.getAssignments().entrySet()) {
+            if (!(projectAssignment.getValue() instanceof SymbolReference)) {
+                return Optional.empty();
+            }
+
+            Symbol outputSymbol = projectAssignment.getKey();
+            Symbol intermediateSymbol = new Symbol(((SymbolReference) projectAssignment.getValue()).getName());
+
+            newProjectionAssignments.put(outputSymbol, outputSymbol.toSymbolReference());
+            if (!newScanOutputs.add(outputSymbol)) {
+                // Duplicate
+                return Optional.empty();
+            }
+            newScanAssignments.put(outputSymbol, assignments.get(intermediateSymbol));
```

Comment:
There is an assumption (global? in base-jdbc?) that table scan's assignments do not rename columns.
Currently the PR hits this:
https://github.com/prestosql/presto/blob/ffd54c918ba7408606687eab5c313435cd98e0a1/presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcRecordSetProvider.java#L54

@martint does it mean this is generally a wrong direction?